src/Streamly/Internal/Data/Stream/Async/Channel/Type.hs

```diff
@@ -84,7 +84,7 @@ data Channel m a = Channel
     , remainingWork  :: Maybe (IORef Count)
     , yieldRateInfo  :: Maybe YieldRateInfo
 
-    , enqueue        :: (RunInIO m, Stream m a) -> IO ()
+    , enqueue        :: Bool -> (RunInIO m, Stream m a) -> IO ()
```

Comment:
We are leaking the inner details of the queue here.

Reply:
That can be made part of the design of the queue having two components, inner and outer. We can think about this later.